### ISSUE TITLE
Adding a PID block with dynamic gains

### ISSUE THREAD

**Issue by SenHuang19** [lbl-srg/modelica-buildings#2993]
This issue is to implement a PID block in which the parameters k, Ti, and Td are inputs.
In addition, the sub-block to approximate the derivative of the input will be modified so that the time constant is an input.
This block will reuse <code>Buildings.Controls.OBC.CDL.Continuous.PIDWithReset</code> and will be added to <code>Buildings.Controls.OBC.Utilities</code>.

**Comment by SenHuang19** [comment]
@mwetter 

In the documentation of <code>Buildings.Controls.OBC.CDL.Continuous.PIDWithReset</code>, it mentions that 

> The derivative of the control error <i>d &frasl; dt e(t)</i> is approximated using
> </p>
> <p align="center" style="font-style:italic;">
> d&frasl;dt x(t) = (e(t)-x(t)) T<sub>d</sub> &frasl; N<sub>d</sub>,
> </p>

Should it be 

 d&frasl;dt x(t) = (e(t)-x(t))  N<sub>d</sub> &frasl; T<sub>d</sub>?

**Comment by mwetter** [comment]
@SenHuang19 : You are right, the fraction is reversed. I will correct this through https://github.com/lbl-srg/modelica-buildings/issues/2994
Thanks for catching this.

**Comment by mwetter** [comment]
@SenHuang19 : Thanks for the PR (https://github.com/lbl-srg/modelica-buildings/pull/3000). I just noted that for the `Derivative` block, the `der()` operator is needed, and hence this block cannot be implemented as a composite block in CDL. We therefore need to propose to add a new block `CDL.Continuous.Derivative` (e.g., an elementary block), which can then be used in the PID controllers.

I will do so through a new issue (https://github.com/lbl-srg/modelica-buildings/issues/3022), and then we can merge that new code to this branch `issue2993_PID_gainInput` and use the block. I should be able to do this next week (tomorrow is already fully booked).

**Comment by SenHuang19** [comment]
@mwetter: Thank you so much! Will the <code>issue2993_PID_gainInput</code> be merged to <code>issue2884_PID_autotuning</code>?

**Comment by mwetter** [comment]
@SenHuang19 : I added the new block `CDL.Continuous.Derivative` to `Utilities.PIDWithInputGains`.
The reference results for `Controls.OBC.Utilities.Validation.PIDWithInputGains` changed; the block now produces different results which is not correct. I will need to see what the cause is next week unless you see how to correct this.

Note that `CDL.Continuous.Derivative` has fewer inputs than your implementation. The inputs are the same as the parameters that are used in the `Derivative` block from the Modelica Standard Library.

I pushed the latest changes to the branch `issue2993_PID_gainInput`
